dvc/remote/base.py

```diff
@@ -712,17 +719,20 @@ def _checkout_dir(
         for entry in dir_info:
             relative_path = entry[self.PARAM_RELPATH]
             entry_checksum = entry[self.PARAM_CHECKSUM]
-            entry_cache_info = self.checksum_to_path_info(entry_checksum)
+            # entry_cache_info = self.checksum_to_path_info(entry_checksum)
             entry_info = path_info / relative_path
+            self._checkout_file(
```

Comment:
Additional difference between _checkout_file and what we had here before is that it saves the link, which we didn't do before to avoid bloating link table. E.g. for directory of 100K files your current patch will add 100K file links to state db link table, which it didn't previously.

Reply:
Got it. I think I could overcome this by adding `save_link` flag that would default to true. What do you think?

Reply:
@pared sounds good.